### PR TITLE
Add metadata update API communication & other features, docs and tests

### PR DESCRIPTION
**Pull request by MVrachev**
# Description

Add missing metadata update features such as:
- communication with API
- `dry-run` option
- `metadata-url` option using python-tuf to fetch latest root

Additional add docs and tests.

Related to: https://github.com/repository-service-tuf/repository-service-tuf-cli/issues/533

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

**Comment by MVrachev** [comment]
I did my best to follow a commit structure so the history should make sense.

File: repository_service_tuf/cli/admin/update.py [hunk 1]
```diff
@@ -107,9 +200,21 @@ def update(root_in, save) -> None:
...
+        task_id = send_payload(
+            settings=settings,
+            url=URL.METADATA.value,
+            payload=asdict(payload),
+            expected_msg="Metadata update accepted.",
+            command_name="Metadata Update",
+        )
+        task_status(task_id, settings, "Metadata Update status: ")
+
+        console.print("Ceremony done. 🔐 🎉. Root metadata update completed.")
```
[the remark this answers is not in the document]
Comment (MVrachev) [reply repository_service_tuf/cli/admin/update.py:220]: I thought it's a ceremony to do a metadata update.

**Comment by MVrachev** [comment]
Addressed your comment @kairoaraujo.

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

File: repository_service_tuf/cli/admin/sign.py [hunk 2]
```diff
@@ -125,7 +122,7 @@ def sign(
 
     # Make sure user understands that result will be send to the API and if the
     # the user wants something else should use '--dry-run'.
-    if settings.get("SERVER") is None and not dry_run:
+    if not settings.get("SERVER") and not dry_run:
         raise click.ClickException(
             "Either '--api-sever' admin option/'SERVER' in RSTUF config or "
```
Comment (kairoaraujo) [inline repository_service_tuf/cli/admin/sign.py:127]:
```suggestion
            "Either '--api-server' admin option/'SERVER' in RSTUF config or "
```
I see this typo several times :)
Not related to your PR, but I will open a issue

**Review by kairoaraujo — APPROVED** [review]

File: repository_service_tuf/cli/admin/sign.py (hunk 2 above)
Comment (MVrachev) [reply repository_service_tuf/cli/admin/sign.py:127]: Will fix it now.

Comment (MVrachev) [reply repository_service_tuf/cli/admin/sign.py]: Fixe

**Review by kairoaraujo — APPROVED** [review]

**Pull request merged**